src/screens/MainScreen.js

```diff
@@ -176,13 +187,43 @@ class TxHistoryView extends React.Component {
     );
   }
 
+  renderFooter = () => {
+    if (!this.state.loading) return null;
+    return (
+      <View style={{ marginTop: 16 }}>
+        <ActivityIndicator />
+      </View>
+    );
+  };
+
+  loadMoreHistory = async () => {
+    if (!this.state.canLoadMore) {
+      // Already loaded all history
+      return;
+    }
+
+    this.setState({ loading: true });
+    const newHistory = await fetchMoreHistory(
```

Comment:
I feel we could use an async iterator here.

Reply:
Why do you think would be good? In the old facade we already have the history pre-calculated, then yielding one of returning a page (15) is not worse and in the wallet service facade we execute a request to get the history, should we request many and yield only one, or request one by one? Either way I feel we just add complexity to the code without any benefits.

Reply:
It feels natural to let the screen decide how many items it would like to show without diving into the details of how it is loaded from the backend. But I feel we don't have to do it in this PR. Maybe we should just create an issue to discuss it further. What do you think?